synapse/tools/apikey.py

```diff
@@ -0,0 +1,97 @@
+import sys
+import asyncio
+import argparse
+
+import synapse.exc as s_exc
+import synapse.telepath as s_telepath
+
+import synapse.lib.time as s_time
+import synapse.lib.output as s_output
+
+descr = '''
+Add, list, or delete user API keys from a Synapse service.
+'''
+
+def printkey(outp, info, apikey=None):
+    iden = info.get('iden')
+    name = info.get('name')
+    created = info.get('created')
+    updated = info.get('updated')
+    expires = info.get('expires')
+
+    outp.printf(f'Iden: {iden}')
+    if apikey:
+        outp.printf(f'  API Key: {apikey}')
+    outp.printf(f'  Name: {name}')
+    outp.printf(f'  Created: {s_time.repr(created)}')
+    outp.printf(f'  Updated: {s_time.repr(updated)}')
+    if expires:
+        outp.printf(f'  Expires: {s_time.repr(expires)}')
+
+    outp.printf('')
+
+async def main(argv, outp=s_output.stdout):
+
+    pars = argparse.ArgumentParser(prog='apikey', description=descr)
+    pars.add_argument('--svcurl', default='cell:///vertex/storage', help='The telepath URL of the Synapse service.')
+
+    subpars = pars.add_subparsers(dest='action', required=True)
+
+    addp = subpars.add_parser('add', help='Add a user API key.')
+    addp.add_argument('-d', '--duration', type=int, help='The duration of the API key in seconds.')
+    addp.add_argument('-u', '--username', type=str, help='The username to add an API key to (restricted to admins).')
```

Comment:
How come these are restricted to admins if the telepath apis are not?  Was it just that you have to use getUserDefByName?

Reply:
Seems like the tool permissions should match the api permissions if it's really just a helper for those and not doing anything extra (i.e. the api could accept a name or iden), but I won't die on that hill.  Could you at least use getUserInfo() so that a non-admin could use this for their own api keys?

Reply:
I copied `getUserDefByName()` from another tool and didn't see `getUserInfo()` which provides the exact functionality I was trying to implement in the tool. Thanks for the tip! Fixed now.

Reply:
I left the `(restricted to admins only)` in the help message though because a user can still only manage their own API keys if they're not an admin.